internal/service/apigatewayv2/stage_unit_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apigatewayv2
+
+import (
+	"encoding/json"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/apigatewayv2/types"
+	"github.com/aws/smithy-go/ptr"
+	"testing"
+)
+
+type jsonSerializableRouteSettings struct {
+	DataTraceEnabled       bool     `json:"data_trace_enabled"`
+	DetailedMetricsEnabled bool     `json:"detailed_metrics_enabled"`
+	LoggingLevel           string   `json:"logging_level"`
+	ThrottlingBurstLimit   *int32   `json:"throttling_burst_limit"`
+	ThrottlingRateLimit    *float64 `json:"throttling_rate_limit"`
+}
+
+func assertRouteSettingsEqualTo(t *testing.T, expected jsonSerializableRouteSettings, actual any) {
+	serialized, err := json.Marshal(actual)
+
+	if err != nil {
+		t.Fatal("unexpected JSON serialization error")
+
+		return
+	}
+
+	var deSerialized jsonSerializableRouteSettings
+
+	err = json.Unmarshal(serialized, &deSerialized)
+
+	if err != nil {
+		t.Fatal("unexpected JSON deserialization error")
+
+		return
+	}
+
+	if !(expected.DataTraceEnabled == deSerialized.DataTraceEnabled &&
+		expected.DetailedMetricsEnabled == deSerialized.DetailedMetricsEnabled &&
+		expected.LoggingLevel == deSerialized.LoggingLevel &&
+		((expected.ThrottlingBurstLimit == nil &&
+			deSerialized.ThrottlingBurstLimit == nil) ||
+			*(expected.ThrottlingBurstLimit) == *(deSerialized.ThrottlingBurstLimit)) &&
+		((expected.ThrottlingRateLimit == nil &&
+			deSerialized.ThrottlingRateLimit == nil) ||
+			*(expected.ThrottlingRateLimit) == *(deSerialized.ThrottlingRateLimit))) {
+		t.Fatal("expected/actual route settings differ")
+	}
```

Comment:
Does the test suite not have any sort of equality check helpers? This is very error-prone, and something like `assert.Equals(t, expected, actual)` would've saved me a good hour of time